akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -120,10 +119,13 @@ private[http] object HttpServerBluePrint {
       override def onPush(): Unit = grab(in) match {
         case RequestStart(method, uri, protocol, hdrs, entityCreator, _, _) ⇒
           val effectiveMethod = if (method == HttpMethods.HEAD && settings.transparentHeadRequests) HttpMethods.GET else method
-          val effectiveHeaders =
-            if (settings.remoteAddressHeader && remoteAddress.isDefined)
-              headers.`Remote-Address`(RemoteAddress(remoteAddress.get)) +: hdrs
-            else hdrs
```

Comment:
Let's keep it as it was before, it was simpler to read and faster for the case where the header was actually used. The extra once-per-connection slowdown of looking up a single attribute should be negligible.